examples/simon.py

```diff
@@ -0,0 +1,339 @@
+
+                    for i in range(0, len(last_work)):
+                        if (last_work[i] == [0 for l in range(0, len(last_work[0]))]):
+                            cv = False
+                        for j in range(0, len(last_work)):
+                            if (last_work[i] == last_work[j] and i != j):
+                                cv = False
+
+
+                    for i in range(1, len(last_work)+1):
+                      for c in list(itertools.combinations([y[0:len(last_work)] for y in last_work], i)):
+                        hol = []
+                        for b in c:
+                          hol.append(b)
+                        calc = [sum(x)%2 for x in zip(*hol)]
+                        ha = True
+                        for ij in range (0, len(hol)):
+                            for ik in range (0, len(hol)):
+                                if (hol[ik] == hol[ij] and ik != ij):
+                                    ha = False
+                        if (ha == True and calc == [0 for p in range(0, len(calc))]):
+                          cv = False
+
+                    #Check if the matrix can be reduced
+
+                    for i in range(0, len(last_work)):
+                        for j in range(i+1, len(last_work)):
+                            if (last_work[i][i] == 0):
+                                cv = False
+                            else:
+                                x = -1*last_work[j][i]/last_work[i][i]
+                                iterator = map(lambda y: y*int(x), last_work[i])
+                                new = [sum(z)%2 for z in zip(last_work[j], iterator)]
+                                last_work[j] = new
+
+                    if (cv == True):
+                        break;
+
+
+        if (cv == True):
+            break;
+
+    matrix = last_work
+
+    return last_work
+
+def construct_solve(matrix_out):
+
+    final_matrix = matrix_out
+
+    solution = []
+
+    for i in range(len(final_matrix)-1, 0, -1):
+      solution.append(final_matrix[i][len(final_matrix[i])-1])
+      for j in range(0, i):
+        if (final_matrix[j][i] == 1):
+          final_matrix[j][len(final_matrix[i])-1] = (final_matrix[j][len(final_matrix[i])-1]-final_matrix[i][len(final_matrix[i])-1])%2
+    solution.append(final_matrix[0][len(final_matrix[i])-1])
+
+    solution.reverse()
+
+    return solution
+
+
+other_matrix_input = copy.deepcopy(matrix_input)
+
+first_shot = shuffle_op(matrix_input, 0)
+try_1 = construct_solve(first_shot)
+second_shot = shuffle_op(other_matrix_input, 1)
+try_2 = construct_solve(second_shot)
+
+processing1 = ''.join(str(x) for x in try_1)
+processing2 = ''.join(str(x) for x in try_2)
+
+the_last = 0
+if (secret_function[1][secret_function[0].index(0)] == secret_function[1][secret_function[0].index(int(processing1, 2))] and secret_function[1][secret_function[0].index(0)] == secret_function[1][secret_function[0].index(int(processing2, 2))]):
```

Comment:
This statement is too long. Create an appropriately named and commented procedure that does this.